### PR TITLE
Activate stateful menu items with right-click without closing the menu

### LINKED ISSUE #10255: Context menu should not close when clicked
**Describe the new feature:** 
When I click on the desired collection, but want to add it to another collection, I have to open the context menu again because it closes on every click. It is necessary to add the ability not to close to him. I do not consider this a bug, so it will be as a suggestion.
Or you can do it by holding down the Ctrl key.
https://youtu.be/bHRd6HTGNpo

### PR DESCRIPTION
**Pull request by 424ever**
- Closes https://github.com/ppy/osu/issues/10255

This is a feature I first saw implemented in [SerenityOS](https://github.com/SerenityOS/serenity). It is for example usefull, when the user wants to add a beatmap to multiple collections in a multiplayer lobby.
[menu.webm](https://github.com/ppy/osu/assets/37748408/2b3001e5-9f7b-4a13-b952-b5991b5e7bef)

### CODE REVIEW

**Comment by peppy** [comment]
I'd prefer this behaviour when right clicking (or adding that in addition to ctrl-click). I don't know where I've seen this done but it feels more natural to me.

**Comment by 424ever** [comment]
Searching for right click I also found this issue talking about the same thing:
#10255 
Where it seemingly was already decided to be handled via right click.

**Comment by peppy** [comment]
The way this manages to work is a touch haphazard (detailed in inline comment) but I'm willing to let it slide just to get this behaviour in without framework changes.

Will wait for one more review just in case there's any strong objections.

**Review by peppy — APPROVED** [review]

**Review by bdach — APPROVED** [review]

**Pull request merged**
